pandas/core/indexing.py

```diff
@@ -1739,22 +1736,26 @@ def _getitem_scalar(self, key):
         values = self.obj._get_value(*key)
         return values
 
-    def _get_partial_string_timestamp_match_key(self, key, labels):
+    def _get_partial_string_timestamp_match_key(self, key, labels: Index):
         """Translate any partial string timestamp matches in key, returning the
         new key (GH 10331)"""
         if isinstance(labels, MultiIndex):
             if isinstance(key, str) and labels.levels[0].is_all_dates:
                 # Convert key '2016-01-01' to
                 # ('2016-01-01'[, slice(None, None, None)]+)
-                key = tuple([key] + [slice(None)] * (len(labels.levels) - 1))
+                key = tuple(
+                    [key] + [slice(None)] * (len(labels.levels) - 1)
```

Comment:
The more I think about this a better idiom would be something like:

```python
new_key = None # type: List[str, slice]
if isinstance(labels, MultiIndex):
   ...
   new_key = tuple([key] + [slice(None)] * len(labels.levels) - 1)
    ...
```

Should appease mypy and prevent the shadowing of the `key` variable